### ISSUE TITLE
Execute Console Messages and Errors from Website

### ISSUE THREAD

**Issue by michaelgohl** [scrapinghub/splash#534]
Hey Guys,

is there a way to execute the console messages and errors with lua_script? Could you provide me an short example? 

How can I enable JS on website rendering?

**Comment by kmike** [comment]
Sorry, what do you mean by "execute console messages and errors"? 

JS rendering is enabled by default, but sometimes there are issues with specific websites because the engine is not new, and because it has Private Mode enabled by default. This is hard to tell without checking error messages.

**Comment by michaelgohl** [comment]
In PhantomJS I am able to see if an JavaScript produces an error or an console message. I want to log them. 

page.onConsoleMessage = function(msg, lineNum, sourceId) { };
page.onError = function(msg, trace) { };

How can I disable private mode?

**Comment by kmike** [comment]
Console capturing is implemented for render.json endpoint (http://splash.readthedocs.io/en/stable/api.html#arg-console), but unfortunately it is not implemented for /execute yet.

**Comment by michaelgohl** [comment]
Message and Error?

**Comment by kmike** [comment]
JS errors in scripts executed by user are converted to Lua errors; third-party errors are logged.

**Comment by michaelgohl** [comment]
Hey,

I used the following curl call:

curl -X POST -H 'content-type: application/json' -d '{"console": 1, "script": 1, "url": "http://box.emsmobile.de/micha/testmicha.html", "wait": 10, "har": 0}' 'http://localhost:8050/render.json'

I get this result:
`{"geometry": [0, 0, 1024, 768], "url": "http://box.emsmobile.de/micha/testmicha.html", "requestedUrl": "http://box.emsmobile.de/micha/testmicha.html", "title": ""}`

The page generates one JavaScript Error and one console message but I didn't see anyone.

**Comment by michaelgohl** [comment]
The Problem is that splash only logs JavaScript Messages. 

In qwebpage.py
`    def javaScriptConsoleMessage(self, msg, line_number, source_id):
        if self.verbosity >= 2:
            log.msg("JsConsole(%s:%d): %s" % (source_id, line_number, msg), system='render')`

Why it doesn't return it to the json? Could you fix this?

**Comment by kmike** [comment]
Hey @michaelgohl,

Your pull requests looks like a good start 👍 To merge a new feature to Splash we usually require docs and tests; it also helps if there is a single pull request for a feature. Are you planning to continue the work on this feature?

**Comment by michaelgohl** [comment]
@kmike Yeah I planning to continue the work on this feature. I need it ;-)
